clouddriver-aws/src/main/groovy/com/netflix/spinnaker/clouddriver/aws/deploy/handlers/LoadBalancerV2UpsertHandler.groovy

```diff
@@ -44,45 +52,72 @@ class LoadBalancerV2UpsertHandler {
   }
   private static String modifyTargetGroupAttributes(AmazonElasticLoadBalancing loadBalancing, LoadBalancer loadBalancer, TargetGroup targetGroup, UpsertAmazonLoadBalancerV2Description.Attributes attributes, DeployDefaults deployDefaults) {
 
-    def currentTargetGroupAttributes = loadBalancing.describeTargetGroupAttributes(new DescribeTargetGroupAttributesRequest()
-      .withTargetGroupArn(targetGroup.targetGroupArn)).attributes
-
     def targetGroupAttributes = []
-    if (attributes) {
-      if (TargetTypeEnum.Lambda.toString().equalsIgnoreCase(targetGroup.getTargetType()))
-      {
-        if (attributes.multiValueHeadersEnabled != null) {
-          targetGroupAttributes.add(new TargetGroupAttribute(key: "lambda.multi_value_headers.enabled", value: attributes.multiValueHeadersEnabled))
-        }
-      } else {
-        Integer deregistrationDelay = [attributes.deregistrationDelay, deployDefaults?.loadBalancing?.deregistrationDelayDefault].findResult(Closure.IDENTITY)
-        if (deregistrationDelay != null) {
-          targetGroupAttributes.add(new TargetGroupAttribute(key: "deregistration_delay.timeout_seconds", value: deregistrationDelay.toString()))
-        }
-        if (loadBalancer.type == 'application') {
-          if (attributes.stickinessEnabled != null) {
-            targetGroupAttributes.add(new TargetGroupAttribute(key: "stickiness.enabled", value: attributes.stickinessEnabled.toString()))
+
+    //Create Target Group Attributes with values provided in description, set to defaults other wise
+    if(deployDefaults !=null) {
```

Comment:
why is this conditional on deployDefaults?

Reply:
so from what i see creation of TargetGroup calls `modifyTargetGroupAttributes` with the deployDefaults and updateTargetGroups calls without , i used that to differentiate between create vs modify 
Modify Target Group -> https://github.com/spinnaker/clouddriver/blob/master/clouddriver-aws/src/main/groovy/com/netflix/spinnaker/clouddriver/aws/deploy/handlers/LoadBalancerV2UpsertHandler.groovy#L201

Create Target Group -> https://github.com/spinnaker/clouddriver/blob/master/clouddriver-aws/src/main/groovy/com/netflix/spinnaker/clouddriver/aws/deploy/handlers/LoadBalancerV2UpsertHandler.groovy#L148

Do you think that's reasonable  approach ?

Reply:
I feel like adding a specific flag to indicate that it is an update vs a create would be better - deployDefaults being null doesn't feel like a super reliable contract to base this behaviour off of and feels ripe for a regression